homeassistant/components/aurora/binary_sensor.py

```diff
@@ -1,146 +1,75 @@
-
-        if self.aurora_data:
-            attrs["visibility_level"] = self.aurora_data.visibility_level
-            attrs["message"] = self.aurora_data.is_visible_text
-            attrs[ATTR_ATTRIBUTION] = ATTRIBUTION
-        return attrs
-
-    def update(self):
-        """Get the latest data from Aurora API and updates the states."""
-        self.aurora_data.update()
-
-
-class AuroraData:
-    """Get aurora forecast."""
-
-    def __init__(self, latitude, longitude, threshold):
-        """Initialize the data object."""
-        self.latitude = latitude
-        self.longitude = longitude
-        self.headers = {USER_AGENT: HA_USER_AGENT}
-        self.threshold = int(threshold)
-        self.is_visible = None
-        self.is_visible_text = None
-        self.visibility_level = None
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    def update(self):
-        """Get the latest data from the Aurora service."""
-        try:
-            self.visibility_level = self.get_aurora_forecast()
-            if int(self.visibility_level) > self.threshold:
-                self.is_visible = True
-                self.is_visible_text = "visible!"
-            else:
-                self.is_visible = False
-                self.is_visible_text = "nothing's out"
-
-        except requests.exceptions.HTTPError as error:
-            _LOGGER.error("Connection to aurora forecast service failed: %s", error)
-            return False
-
-    def get_aurora_forecast(self):
-        """Get forecast data and parse for given long/lat."""
-        raw_data = requests.get(URL, headers=self.headers, timeout=5).text
-        # We discard comment rows (#)
-        # We split the raw text by line (\n)
-        # For each line we trim leading spaces and split by spaces
-        forecast_table = [
-            row.strip().split()
-            for row in raw_data.split("\n")
-            if not row.startswith("#")
-        ]
-
-        # Convert lat and long for data points in table
-        # Assumes self.latitude belongs to [-90;90[ (South to North)
-        # Assumes self.longitude belongs to [-180;180[ (West to East)
-        # No assumptions made regarding the number of rows and columns
-        converted_latitude = floor((self.latitude + 90) * len(forecast_table) / 180)
-        converted_longitude = floor(
-            (self.longitude + 180) * len(forecast_table[converted_latitude]) / 360
-        )
-
-        return forecast_table[converted_latitude][converted_longitude]
+        return {"attribution": ATTRIBUTION}
+
+    @property
+    def icon(self):
+        """Return the icon for the sensor."""
+        return "mdi:hazard-lights"
+
+    @property
+    def device_info(self):
+        """Define the device based on name."""
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self._unique_id)},
+            ATTR_NAME: self.coordinator.name,
+            ATTR_MANUFACTURER: "NOAA",
+            ATTR_MODEL: "Aurora Visibility Sensor",
+        }
```

Comment:
If this is a service we should return that as entry type here.

https://developers.home-assistant.io/docs/device_registry_index#device-properties

Reply:
Addressed in PR https://github.com/home-assistant/core/pull/43148